database/src/main/resources/db/migration/V64__add_vedtak_table.sql

```diff
@@ -0,0 +1,149 @@
+create table if not exists vedtak
+(
+    id                        uuid        not null primary key,
+    opprettet                 timestamptz not null,
+    fraOgMed                  date,
+    tilOgMed                  date,
+    saksbehandler             text        not null,
+    attestant                 text        not null,
+    behandlingsinformasjon    jsonb       not null,
+    utbetalingid              varchar(30) references utbetaling(id),
+    simulering                jsonb,
+    beregning                 jsonb,
+    iverksattjournalpostid    text,
+    iverksattBrevbestillingId text
+);
+
+create table if not exists behandling_vedtak
+(
+    id                  uuid primary key,
+    vedtakId            uuid references vedtak (id) NOT NULL,
+    sakId               uuid references sak (id)    NOT NULL,
+    søknadsbehandlingId uuid references behandling (id),
+    revurderingId       uuid references revurdering (id),
+
+    CONSTRAINT revurderingId_eller_søknadsbehandlingId
+        CHECK (
+                (søknadsbehandlingId IS NULL AND revurderingId IS NOT NULL)
+                OR
+                (søknadsbehandlingId IS NOT NULL AND revurderingId IS NULL)
+            )
+);
+
+-- Populer vedtaktabellen (og knytningstabellen) med eksisterende søknadsbehandlinger og revurderinger
+WITH b AS (
+    SELECT *, uuid_generate_v4() vedtakId
+    FROM behandling
+    WHERE status IN ('IVERKSATT_INNVILGET', 'IVERKSATT_AVSLAG')
+),
+     v as (
+         INSERT INTO vedtak (
+                             id,
+                             opprettet,
+                             fraOgMed,
+                             tilOgMed,
+                             saksbehandler,
+                             attestant,
+                             behandlingsinformasjon,
+                             utbetalingid,
+                             simulering,
+                             beregning,
+                             iverksattjournalpostid,
+                             iverksattBrevbestillingId
+             ) (SELECT b.vedtakId,
+                       NOW(),
```

Comment:
Blir det ikke mer korrekt med b.opprettetId?

Reply:
Nja, det blir feil det også, siden `b.opprettet` er tidspunktet behandlingen ble opprettet på; ikke når den ble iverksatt. Så vi tenkte iallfall da at siden det uansett blir feil så kan de iallfall settes til et tidspunkt som gjør at vi ser hvilke vedtak som kom fra migreringen. Kunne kanskje ha brukt en eller annen `MIN_VALUE` i stedet, men dette er tidspunkt som potensielt kan brukes i visning eller sortering eller utplukking (finn vedtak innenfor intervall), og da ville `MIN_VALUE` ha truffet på litt for mange caser :p

Reply:
Godt poeng, vi lagrer ingen tilstandsoverganger i behandlingstabellen. Da er sikkert NOW() tilstrekkelig, hvis ikke så kunne man hentet den inn fra vedtakssnapshot :shrug:

Reply:
Oh, det vakke dumt! Vi kan undersøke det.